platform/admin/configurator/admin-configurator-api/src/main/java/org/codice/ddf/admin/configurator/OperationReport.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator;
+
+import java.util.List;
+
+/**
+ * <b> This code is experimental. While this class is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface OperationReport {
+    boolean isTransactionSucceeded();
+
+    Result getResult(String key);
+
+    /**
+     * An immutable list of Results that failed.
+     *
+     * @return an immutable list of failed results.
+     * It is an implementor's responsibility to ensure that the collection is unmodifiable.
+     */
+    List<Result> getFailedResults();
+
+    boolean containsFailedResults();
+
+    void putResult(String key, Result result);
```

Comment:
I think the increased complexity to the Configurator implementation provided would be pretty bad. It's not undoable, but I'm not sure it it's a valuable enough change.